src/sentry/api/bases/sentryapps.py

```diff
@@ -99,6 +103,45 @@ def dispatch(self, request, *args, **kwargs):
 class SentryAppsBaseEndpoint(IntegrationPlatformEndpoint):
     permission_classes = (SentryAppsPermission,)
 
+    def _get_organization_slug(self, request):
+        organization_slug = request.json_body.get("organization")
+        if not organization_slug or not isinstance(organization_slug, string_types):
+            error_message = """
+                Please provide a valid value for the 'organization' field.
+            """
+            raise ValidationError({"organization": to_single_line_str(error_message)})
+        return organization_slug
+
+    def _get_organization_for_superuser(self, organization_slug):
+        try:
+            return Organization.objects.get(slug=organization_slug)
+        except Organization.DoesNotExist:
+            error_message = """
+                Organization '{}' does not exist.
+            """.format(
+                organization_slug
+            )
+            raise ValidationError({"organization": to_single_line_str(error_message)})
+
+    def _get_organization_for_user(self, user, organization_slug):
+        try:
+            return user.get_orgs().get(slug=organization_slug)
+        except Organization.DoesNotExist:
+            error_message = """
+                User does not belong to the '{}' organization.
+            """.format(
+                organization_slug
+            )
+            raise PermissionDenied(to_single_line_str(error_message))
```

Comment:
@jverce We probably want a `ResourceDoesNotExist` error here instead so we don't give the impression that the organization might exist.

Reply:
It's up to you honestly (you know the product much better than me) so I can definitely change it, but it feels strange to me that a `POST` endpoint returns a 404 error. WDYT?

Reply:
@jverce Yea, it is kind of weird. This endpoint really should have had the org slug in the route itself instead of the post (which would have made the 404 make more sense) but it's hard to change that now when people are using the APIs already.

Reply:
Maybe a generic `400 Bad Request` with something like "invalid organization"?

Reply:
@jverce We can keep the `403` if that still makes sense to you after my explanation.